CI failing when disabling tokenization

We run some tests every day, instead of on every PR
One of the [test is to ensure we can disable the tokenization](https://github.com/meilisearch/meilisearch/blob/29adfc2f6835fb17a584f5e2bcf7d7acde48e44b/.github/workflows/test-suite.yml#L108-L129) (when compiling Meilisearch) without any issue.
Currently, our CI testing this is failing every day (not randomly).

It seems like disabling all the features, i.e. `no-default-features`, of Meilisearch no longer removes the lindera-related libraries from Meilisearch and continues to compile them. The lindera libraries are used by the direct dependency charabia. The main goal of disabling lindera was to reduce the binary size and the compilation time.

You should probably take a look at [the charabia _Cargo.toml_](https://github.com/meilisearch/charabia/blob/de62ab9b889061126b0a8473aa53bf6288a99679/charabia/Cargo.toml#L27C1-L38) and make some of the libraries optional.

I tried to delete all the GitHub Actions cache, but nothing changed, and the CI is still failing
https://github.com/meilisearch/meilisearch/actions/runs/5817159170/job/15771397666

```
  ...
  Downloaded hmac v0.12.1
  Downloaded generic-array v0.14.7
  Downloaded futures-executor v0.3.28
  Downloaded futures-macro v0.3.28
  Downloaded serde_urlencoded v0.7.1
  Downloaded cfg-if v1.0.0
  Downloaded bytes v1.4.0
  Downloaded either v1.8.1
Error: Process completed with exit code 1.
```

The main problem is the error is really vague.

--- a/.github/workflows/test-suite.yml
+++ b/.github/workflows/test-suite.yml
@@ -123,7 +123,9 @@ jobs:
           override: true
       - name: Run cargo tree without default features and check lindera is not present
         run: |
-          cargo tree -f '{p} {f}' -e normal --no-default-features | grep lindera -vqz
+          if ! cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
+            echo "lindera not found"
+          fi
       - name: Run cargo tree with default features and check lindera is pressent
         run: |
           cargo tree -f '{p} {f}' -e normal | grep lindera -qz

--- a/.github/workflows/test-suite.yml
+++ b/.github/workflows/test-suite.yml
@@ -124,7 +124,7 @@ jobs:
       - name: Run cargo tree without default features and check lindera is not present
         run: |
           if cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
-            echo "lindera found"
+            echo "lindera has been found in the sources and it shouldn't"
           fi
       - name: Run cargo tree with default features and check lindera is pressent
         run: |

--- a/.github/workflows/test-suite.yml
+++ b/.github/workflows/test-suite.yml
@@ -125,6 +125,7 @@ jobs:
         run: |
           if cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
             echo "lindera has been found in the sources and it shouldn't"
+            exit 1
           fi
       - name: Run cargo tree with default features and check lindera is pressent
         run: |

--- a/.github/workflows/test-suite.yml
+++ b/.github/workflows/test-suite.yml
@@ -123,8 +123,8 @@ jobs:
           override: true
       - name: Run cargo tree without default features and check lindera is not present
         run: |
-          if ! cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
-            echo "lindera not found"
+          if cargo tree -f '{p} {f}' -e normal --no-default-features | grep -vqz lindera; then
+            echo "lindera found"
           fi
       - name: Run cargo tree with default features and check lindera is pressent
         run: |

